genie-web/src/main/java/com/netflix/genie/web/tasks/leader/ClusterCheckerTask.java

```diff
@@ -109,70 +121,99 @@ public ClusterCheckerTask(
     @Override
     public void run() {
         log.info("Checking for cluster node health...");
-        final Set<String> badNodes = new HashSet<>();
         this.jobSearchService.getAllHostsWithActiveJobs()
             .stream()
             .filter(host -> !this.hostName.equals(host))
-            .forEach(
-                host -> {
-                    try {
-                        restTemplate.getForObject(this.scheme + host + this.healthEndpoint, String.class);
-                    } catch (final Exception ioe) {
-                        log.error("Unable to reach {}", host, ioe);
-                        badNodes.add(host);
-                    }
+            .forEach(this::validateHostAndUpdateErrorCount);
+
+        this.errorCounts.entrySet().removeIf(entry -> {
+            final String host = entry.getKey();
+            boolean result = true;
+            if (entry.getValue() >= properties.getLostThreshold()) {
+                try {
+                    updateJobsToFailedOnHost(host);
+                } catch (Exception e) {
+                    log.error("Unable to update jobs on host {} due to exception", host, e);
+                    unableToUpdateJobCounter.increment();
+                    result = false;
                 }
-            );
+            } else {
+                result = false;
```

Comment:
Always want to have the return at the end. This logic will remove the entry if it returns true.